src/rgw/rgw_bucket_logging.cc

```diff
@@ -183,6 +185,28 @@ std::string unique_string() {
   return str;
 }
 
+// create an incremental string of N characters if possible
+// fallback to a random string of N characters if not
+template<size_t N>
+std::string incremental_string(const DoutPrefixProvider *dpp, std::optional<std::string> old_name) {
+  // for the fist time we create a string of zeros
+  if (!old_name) {
+    return std::string(N, '0');
+  }
+  const auto str_counter = old_name->substr(old_name->length() - N+1, N);
+  try {
+    auto counter = boost::lexical_cast<unsigned long>(str_counter);
+    ++counter;
+    std::ostringstream ss;
```

Comment:
is there a way to use a mechanism other than stringstream here?  we try very hard to avoid it for efficiency reasons

Reply:
i.e., std/fmt::format I believe